src/Data/List.purs

```diff
@@ -712,10 +711,10 @@ instance functorList :: Functor List where
 instance foldableList :: Foldable List where
   foldr _ b Nil = b
   foldr o b (Cons a as) = a `o` foldr o b as
-  foldl _ b Nil = b
-  foldl o b (Cons a as) = foldl o (b `o` a) as
-  foldMap _ Nil = mempty
-  foldMap f (Cons x xs) = f x <> foldMap f xs
+  foldl = go
+    where go _ b Nil = b
+          go o b (Cons a as) = go o (b `o` a) as
+  foldMap f = foldl (\acc -> append acc <<< f) mempty
```

Comment:
This is left-to-right now, but was right-to-left before. For non-commutative monoids, that might be an issue.